internal/app/repository/user_repository.go

```diff
@@ -36,3 +39,22 @@ func (r *UserRepository) GetUserByUsernameAndPassword(ctx context.Context, usern
 
 	return &user, nil
 }
+
+func (r *UserRepository) RegisterUser(ctx context.Context, username string, hashPassword string, Email string, Name string) (bool, error) {
```

Comment:
i think you need to return ID user you just inserted on this RegisterUser function.